src/main/flight/gps_rescue.c

```diff
@@ -120,31 +121,40 @@ typedef struct {
 #define GPS_RESCUE_RATE_SCALE_DEGREES    45 // Scale the commanded yaw rate when the error is less then this angle
 #define GPS_RESCUE_SLOWDOWN_DISTANCE_M  200 // distance from home to start decreasing speed
 
-PG_REGISTER_WITH_RESET_TEMPLATE(gpsRescueConfig_t, gpsRescueConfig, PG_GPS_RESCUE, 1);
-
-PG_RESET_TEMPLATE(gpsRescueConfig_t, gpsRescueConfig,
-    .angle = 32,
-    .initialAltitudeM = 50,
-    .descentDistanceM = 200,
-    .rescueGroundspeed = 2000,
-    .throttleP = 150,
-    .throttleI = 20,
-    .throttleD = 50,
-    .velP = 80,
-    .velI = 20,
-    .velD = 15,
-    .yawP = 40,
-    .throttleMin = 1200,
-    .throttleMax = 1600,
-    .throttleHover = 1280,
-    .sanityChecks = RESCUE_SANITY_ON,
-    .minSats = 8,
-    .minRescueDth = 100,
-    .allowArmingWithoutFix = false,
-    .useMag = true,
-    .targetLandingAltitudeM = 5,
-    .targetLandingDistanceM = 10,
-);
+PG_REGISTER_WITH_RESET_FN(gpsRescueConfig_t, gpsRescueConfig, PG_GPS_RESCUE, 1);
+
+void pgResetFn_gpsRescueConfig(gpsRescueConfig_t *config)
+{
+    gpsRescueConfig_t resetConfig = {
```

Comment:
Should be const really.

Reply:
@mikeller 
If we define a constant, say, `GPS_RESCUE_USE_MAG_DEFAULT` to `true` or `false` based on `USE_MAG`, then we can use `PG_REGISTER_WITH_RESET`.  What do you think?

Reply:
@jflyper: Definitely. It's done in other places, so why not here.

Reply:
👍

Reply:
Done